change_notes/2024-10-08-upgrade-to-2.16.6.md

```diff
@@ -0,0 +1,3 @@
+- Updated the CodeQL version to `2.16.6`.
+- `M0-1-2` - `InfeasiblePath.ql`:
+  - This query may now report additional results within templates where a relational operation is performed which has a constant value given the specified arguments.
```

Comment:
if we do think the results will change, can we add a testcase to help demonstrate the scenario where we expect a change?

Reply:
Good point. I've added a test case which includes a relational operation that wouldn't have been found by the previous implementation.